rider-core/src/test/java/com/github/database/rider/core/DataSetProviderIt.java

```diff
@@ -0,0 +1,206 @@
+        //assertThat(count).isEqualTo(1); //assertion in expectedDataSet
+    }
+
+    @Test
+    public void shouldSeedDataSetUsingClassLevelDataSetProvider() {
+        Tweet tweet = (Tweet) EntityManagerProvider.em().createQuery("select t from Tweet t where t.id = 'abcdef12345'").getSingleResult();
+        assertThat(tweet).isNotNull()
+            .extracting("content")
+            .contains("dbrider rules!");
+    }
+
+    @Test
+    @DataSet(provider = CompositeDataSetProvider.class)
+    public void shouldSeedDataSetUsingCompositeDataSetProvider() {
+        List<User> users = EntityManagerProvider.em().createQuery("select u from User u ").getResultList();
+        assertThat(users).
+            isNotNull().
+            isNotEmpty().hasSize(2).
+            extracting("name").
+            contains("@dbunit", "@dbrider");
+        Tweet tweet = (Tweet) EntityManagerProvider.em().createQuery("select t from Tweet t where t.id = 'abcdef12345'").getSingleResult();
+        assertThat(tweet).isNotNull()
+            .extracting("content")
+            .contains("dbrider rules!");
+    }
+
+    @Test
+    public void shouldSeedDatabaseUsingDataSetProviderWithoutAnnotatation() throws SQLException {
+        try (Connection conn = EntityManagerProvider.instance("executor-it").connection()) {
+            DataSetExecutor executor = DataSetExecutorImpl.instance("executor-name", new ConnectionHolderImpl(conn));
+            DataSetConfig DataSetConfig = new DataSetConfig()
+                .datasetProvider(UserDataSetProvider.class)
+                .disableConstraints(true);
+            executor.clearDatabase(DataSetConfig);
+            executor.createDataSet(DataSetConfig);
+            List<User> users = EntityManagerProvider.em("executor-it").createQuery("select u from User u ").getResultList();
+            assertThat(users).
+                isNotNull().
+                isNotEmpty().hasSize(2).
+                extracting("name").
+                contains("@dbunit", "@dbrider");
+        }
+
+    }
+
+    public static class UserDataSetProvider implements DataSetProvider {
+
+        @Override
+        public IDataSet provide() throws DataSetException {
+            RiderDataSetBuilder builder = new RiderDataSetBuilder(true);
+            builder.newRow("user").with("id", 1)
+                .with("name", "@dbunit").add()
+                .newRow("user").with("id", 2)
+                .with("name", "@dbrider").add();
+            return builder.build();
+        }
+    }
+
+    public static class UsersWithBrokenReferentialConstraintProvider implements DataSetProvider {
+
+        @Override
+        public IDataSet provide() throws DataSetException {
+            RiderDataSetBuilder builder = new RiderDataSetBuilder();
+            ColumnSpec<Integer> id = ColumnSpec.newColumn("ID");
+            ColumnSpec<String> name = ColumnSpec.newColumn("NAME");
+            builder.newRow("USER").with("ID", 1)
```

Comment:
May be replace with more DB oriented DSL like row("USER").column("ID", 1).column("NAME", "a")...

Reply:
This is from DBunitDataSetBuilder library, we're reusing its api, see here: https://github.com/opensource21/dbunit-datasetbuilder

Reply:
Here is where we extend DBunitDataSetBuilder API: https://github.com/rmpestano/database-rider/blob/f964f8d36bc9b5e921101ee62f9df90bc663198c/rider-core/src/main/java/com/github/database/rider/core/dataset/builder/RiderDataSetBuilder.java#L23

If anyone have an idea on how to improve e.g use `builder.row().column()` it is welcomed

Reply:
Is it a good idea to use the datasetbuilder though? It hasn't been updated in 6 years. Maybe just copy the idea from there. What I also like about databaserider is that it doesn't introduce many other dependencies.

Reply:
May be instead of extention we could use delegation?

Reply:
Yea delegation could solve it but thinking better I think we can copy it to dbrider so we can evolve the builder easily later. Also it is no a big lib and we can copy only what need.
 
Thank you guys, I'll ping you back when I'm done.

Reply:
Just moved builder to dbrider, [see changes here](https://github.com/database-rider/database-rider/pull/132/commits/14d6ef190a18df283dc37570506619018e4e9034). Any feedback is appreciated.

Reply:
Looks good!

Reply:
In my opinion the original API was actually slightly better. Why so? Because `newRow(”user”).with(”id”, 1)` indicates that we are creating a new row to the user table with a value of 1 for the id.

The new form `row(”user”).column(”id”, 1)` is making these concepts less clear. Is it a row called user? A row doesn’t have a name but a table does. (Actually you could say the same for the original) Why is there a number with the column?

I would suggest that from a fluent API perspective a better alternative would be something like `table(”user”).newRow().withFieldValue(”id”, 1)`.

 Thoughts?

Reply:
For me either way is ok but let's discuss the builder on the issue #129

Reply:
Sure. It might be a matter of preference :) The other one is maybe more simple and the other one more expressive.

Reply:
I also like row/column